File: leetcode/141-linked-list-cycle/main.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
      if (!head || !head->next) {
        return false;
      }

      ListNode *s = head;
      ListNode *f = head->next;

      while (f && f->next) {
        s = s->next;
        f = f->next->next;

        if (s == f) {
          return true;
        }

      }

      return false;
        
    }
};

File: leetcode/141-linked-list-cycle/main.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
	if (head == nil || head.Next == nil) {
		return false;
	}
	
	s := head;
 	f := head.Next;

	for (f != nil && f.Next != nil) {
		s = s.Next;
		f = f.Next.Next;
		
		if (s == f) {
			return true;
		}

	}
	
	return false;
    
}
